src/dbnode/namespace/types.go

```diff
@@ -80,6 +80,12 @@ type Options interface {
 	// ColdWritesEnabled returns whether cold writes are enabled for this namespace.
 	ColdWritesEnabled() bool
 
+	// SetCacheBlocksOnRetrieve sets whether to cache blocks from this namespace when retrieved.
+	SetCacheBlocksOnRetrieve(value bool) Options
```

Comment:
Mind adding a comment to these ones to indicate that this set to true will be overriden if the global cache flag is off?